cloudinit/config/schemas/schema-cloud-config-v1.json

```diff
@@ -2199,21 +2199,37 @@
       "properties": {
         "ubuntu_advantage": {
           "type": "object",
-          "required": ["token"],
           "additionalProperties": false,
           "properties": {
             "enable": {
               "type": "array",
               "items": {"type": "string"},
-              "description": "Optional list of ubuntu-advantage services to enable. Any of: cc-eal, cis, esm-infra, fips, fips-updates, livepatch. By default, a given contract token will automatically enable a number of services, use this list to supplement which services should additionally be enabled. Any service unavailable on a given Ubuntu release or unentitled in a given contract will remain disabled."
+              "description": "Optional list of ubuntu-advantage services to enable. Any of: cc-eal, cis, esm-infra, fips, fips-updates, livepatch. By default, a given contract token will automatically enable a number of services, use this list to supplement which services should additionally be enabled. Any service unavailable on a given Ubuntu release or unentitled in a given contract will remain disabled. In Ubuntu Pro instances, if this list is given, then only those services will be enabled, ignoring contract defaults."
+            },
+            "enable_beta": {
+              "type": "array",
+              "items": {"type": "string"},
+              "description": "Optional list of ubuntu-advantage beta services to enable. By default, a given contract token will automatically enable a number of services, use this list to supplement which services should additionally be enabled. Any service unavailable on a given Ubuntu release or unentitled in a given contract will remain disabled. In Ubuntu Pro instances, if this list is given, then only those services will be enabled, ignoring contract defaults."
             },
             "token": {
               "type": "string",
-              "description": "Required contract token obtained from https://ubuntu.com/advantage to attach."
+              "description": "Contract token obtained from https://ubuntu.com/advantage to attach (in non-Pro instances)."
+            },
+            "features": {
+              "type": "object",
+              "description": "Ubuntu Advantage features.",
+              "additionalProperties": false,
+              "properties": {
+                "disable_auto_attach": {
+                  "type": "boolean",
+                  "description": "Optional boolean for controlling if auto attach (in Ubuntu Pro instances) will be performed. Default: ``false``",
+                  "default": false
+                }
+              }
             },
             "config": {
               "type": "object",
-              "description": "Configuration settings or override Ubuntu Advantage config",
+              "description": "Configuration settings or override Ubuntu Advantage config. Only used in non Ubuntu Pro instances.",
```

Comment:
During the implementation of https://github.com/canonical/ubuntu-advantage-client/pull/2171, it was possible to pass an `UAConfig` instance (containing this config) to [full_auto_attach](https://github.com/canonical/ubuntu-advantage-client/pull/2171/files#diff-4fa660a53fa2584a90fbec8daf677a41bea763aed263b1cf98390656405c850eR65). But this did not get to the merged commit. Is there an alternative way to pass the config? Or is it not necessary to do so in Pro instances? @CalvoM

Reply:
We have a way of passing the UAConfig to the endpoint via [_full_auto_attach](https://github.com/canonical/ubuntu-advantage-client/blob/e2e9fe94d619f2522ee4b18d411d8efe52fc4844/uaclient/api/u/pro/attach/auto/full_auto_attach/v1.py#L66). However, we recommend that you use the endpoint not requiring the UAConfig parameter, since we create the object internally. You can use the other endpoint, when you have some custom configuration for UAConfig default object.